packages/notifier/src/notifier.js

```diff
@@ -0,0 +1,54 @@
+import { producePromise } from '@agoric/produce-promise';
+import harden from '@agoric/harden';
+
+/**
+ * Makes a notifier object, which allows a service to produce a stream of update
+ * promises. Each update is a record containing
+ *      { currentState, nextStatePromise, updateHandle }.
+ *   currentState is whatever state the service wants to publish,
+ *   nextStatePromise is a HandledPromise, which resolves when the state changes
+ *   updateHandle is an opaque object that identifies the update.
+ *
+ * The notifier has two methods: getUpdateSince(handle), and updateState(state).
+ * getUpdateSince() returns the record above, or a Promise for it. If the
+ * argument to getUpdateSince(handle) is omitted or differs from the current
+ * handle, the current record is returned. If the handle matches the current
+ * state, then a promise is returned, which will resolve to the next record
+ * after the state next changes. nextStatePromise in the record is that same
```

Comment:
This is counter-intuitive. Wouldn't it make more sense for `nextStatePromise` to fulfill when there's a state after this one?